packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/LightCurveViewer/d3/addDataMask.spec.js

```diff
@@ -1,7 +1,41 @@
+import * as d3 from 'd3'
+
 import addDataMask from './addDataMask'
 
+const outerMargin = 10
+const id = 1
 describe('LightCurveViewer > d3 > addDataMask', function () {
+  let body, node, d3svg
+  before(function () {
+    body = document.getElementsByTagName('body')[0]
+    node = document.createElement('svg')
+    body.appendChild(node)
+    d3svg = d3.select(node)
+  })
+
+  after(function () {
+    body.removeChild(node)
+  })
+
+  afterEach(function () {
+    node.innerHTML = ''
```

Comment:
This is just for my understanding, no rush to respond, but what is the reason for resetting the innerHTML after each test? Also, in #968 the test svg node was built for each test compared to these tests where it's built with just before/after, and I'm not sure why one or the other.

Reply:
It's because the d3 API being chainable. The `addDataMask` function adds a `clipPath`, then adds a `rect` as a child of `clipPath`. If `addDataMask` is updated to have more nodes added in the chain, I want to be sure that the `innerHTML` of the parent `clipPath` is reset for each test.

The other function in #968 is three `g` elements that are the child of the `svg` so I found it best to just remove the whole `svg` between each test.